src/nvme/tree.c

```diff
@@ -2309,7 +2330,8 @@ static int nvme_ns_init(struct nvme_ns *n)
 	n->lba_util = le64_to_cpu(ns.nuse);
 	n->meta_size = le16_to_cpu(ns.lbaf[flbas].ms);
 
-	if (!nvme_ns_identify_descs(n, descs))
+	if (!(n->c->quirk_flags & NVME_QUIRKS_NS_ID_DESC_LIST) &&
```

Comment:
This doesn't work because `n->c` is not valid yet. The controller is to the namespace after `nvme_ns_init` has been called when getting called from `nvme_scan_topology`. But there are other callers such as `nvme_scan_namespace` which only have an string as argument. 

Hmm, maybe we should just try to get rid of all ioctl calls in the `nvme_scan_topology` call chain. See also https://github.com/linux-nvme/libnvme/issues/681

Reply:
We discussed this during ALPSS and yep, let's ditch all commands from the scan_topology call and completely rely on sysfs.